src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -0,0 +1,581 @@
+  {
+    mz_tolerance_ = mz_tolerance;
+  }
+
+  double TargetedSpectraExtractor::getMZTolerance() const
+  {
+    return mz_tolerance_;
+  }
+
+  void TargetedSpectraExtractor::setMZUnit(const bool& mz_unit_is_Da)
+  {
+    mz_unit_is_Da_ = mz_unit_is_Da;
+  }
+
+  bool TargetedSpectraExtractor::getMZUnit() const
+  {
+    return mz_unit_is_Da_;
+  }
+
+  void TargetedSpectraExtractor::setSGolayFrameLength(const UInt& sgolay_frame_length)
+  {
+    sgolay_frame_length_ = sgolay_frame_length;
+  }
+
+  UInt TargetedSpectraExtractor::getSGolayFrameLength() const
+  {
+    return sgolay_frame_length_;
+  }
+
+  void TargetedSpectraExtractor::setSGolayPolynomialOrder(const UInt& sgolay_polynomial_order)
+  {
+    sgolay_polynomial_order_ = sgolay_polynomial_order;
+  }
+
+  UInt TargetedSpectraExtractor::getSGolayPolynomialOrder() const
+  {
+    return sgolay_polynomial_order_;
+  }
+
+  void TargetedSpectraExtractor::setGaussWidth(const double& gauss_width)
+  {
+    gauss_width_ = gauss_width;
+  }
+
+  double TargetedSpectraExtractor::getGaussWidth() const
+  {
+    return gauss_width_;
+  }
+
+  void TargetedSpectraExtractor::setUseGauss(const bool& use_gauss)
+  {
+    use_gauss_ = use_gauss;
+  }
+
+  bool TargetedSpectraExtractor::getUseGauss() const
+  {
+    return use_gauss_;
+  }
+
+  void TargetedSpectraExtractor::setSignalToNoise(const double& signal_to_noise)
+  {
+    signal_to_noise_ = signal_to_noise;
+  }
+
+  double TargetedSpectraExtractor::getSignalToNoise() const
+  {
+    return signal_to_noise_;
+  }
+
+  void TargetedSpectraExtractor::setPeakHeightMin(const double& peak_height_min)
+  {
+    peak_height_min_ = peak_height_min;
+  }
+
+  double TargetedSpectraExtractor::getPeakHeightMin() const
+  {
+    return peak_height_min_;
+  }
+
+  void TargetedSpectraExtractor::setPeakHeightMax(const double& peak_height_max)
+  {
+    peak_height_max_ = peak_height_max;
+  }
+
+  double TargetedSpectraExtractor::getPeakHeightMax() const
+  {
+    return peak_height_max_;
+  }
+
+  void TargetedSpectraExtractor::setFWHMThreshold(const double& fwhm_threshold)
+  {
+    fwhm_threshold_ = fwhm_threshold;
+  }
+
+  double TargetedSpectraExtractor::getFWHMThreshold() const
+  {
+    return fwhm_threshold_;
+  }
+
+  void TargetedSpectraExtractor::setTICWeight(const double& tic_weight)
+  {
+    tic_weight_ = tic_weight;
+  }
+
+  double TargetedSpectraExtractor::getTICWeight() const
+  {
+    return tic_weight_;
+  }
+
+  void TargetedSpectraExtractor::setFWHMWeight(const double& fwhm_weight)
+  {
+    fwhm_weight_ = fwhm_weight;
+  }
+
+  double TargetedSpectraExtractor::getFWHMWeight() const
+  {
+    return fwhm_weight_;
+  }
+
+  void TargetedSpectraExtractor::setSNRWeight(const double& snr_weight)
+  {
+    snr_weight_ = snr_weight;
+  }
+
+  double TargetedSpectraExtractor::getSNRWeight() const
+  {
+    return snr_weight_;
+  }
+
+  void TargetedSpectraExtractor::updateMembers_()
```

Comment:
this just seems like a bad idea as you provide two interfaces with which to change the members of the class:

- through the regular Param class
- through all the get/set methods

I suggest to use only one method